// hi-mom/lib/hi-mom.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var languages = {
  ar: "مرحباً, {{امي}}!",
  "ar-IQ": "هلا, {{يمه}}!",
  as: "মা, {{অ' মা}}",
  "ar-MA": "سلام، {{ماما}}",
  az: 'Salam, {{ana}}!',
  ca: "Hola, {{mamà}}!",
  cn: "你好，{{妈妈}}!",
  cs: "Ahoj, {{mamá}}!",
  de: "Hallo, {{Mama}}!",
  en: "Hi, {{mom}}!",
  "en-AU": "Hi, {{mum}}!",
  "en-PT": "Ahoy, {{mom}}!",
  es: "Hola, {{mamá}}!",
  fa: "سلام، {{مامان}}!",
  fr: "Salut, {{maman}}!",
  he: "שלום, {{מה}}!",
  hi: "नमस्ते {{मम्मी}}!",
  hu: "Szia, {{anya}}!",
  id: "Hai, {{ibu}}!",
  it: "Ciao {{mamma}}!",
  ja: "こんにちは、{{ママ}}!",
  ko: "안녕하세요, {{마모}}!",
  lt: "Labas, {{mama}}!",
  ms: "Hai, {{bapa}}!",
  "nl-BE": "Hallo, {{moeke}}!",
  "nl-NL": "Hallo, {{moeder}}!",
  pl: "Cześć, {{matka}}!",
  pt: "Oi, {{mãe}}!",
  ro: "Bună, {{mamă}}!",
  ru: "Привет, {{мама}}!",
  si: "ආයුබෝවන්, {{අම්මේ}}!",
  sr: "Ćao, {{mama}}!",
  ta: "வணக்கம் {{அம்மா}}",
  th: "สวัสดีค่ะ, {{มา}}!",
  tr: "Merhaba, {{anne}}!",
  tm: "Salam, {{eje}}!",
  ua: "Привіт, {{мама}}!",
  ur: "سلام، {{امی}}",
  vi: "Xin chào, {{mẹ}}!",
  zh: "你好，{{妈妈}}!",
};

/**
 * A blazingly fast way to say hi to your mom!
 * @param {string} name - The mom name
 * @param {string} [language="en"] - The language
 * @example hiMom()
 * // "Hi, mom!"
 * @example hiMom("Kate")
 * // "Hi, Kate!"
 * @example hiMom("Maria", "pt")
 * // "Oi, Maria!"
 * @returns {string} A greeting to your mom in the informed language
 */
function hiMom(name, language = "en") {
  if (!language || !languages.hasOwnProperty(language)) {
    throw new Error("Language not yet supported, but hi mom anyway!");
  }

  const regex = /\{{2}([^}]+)\}{2}/;
  const greeting = languages[language];
  const localeName = greeting.split(regex)[1];

  return greeting.replace(regex, name || localeName);
}

function hiMoms(mothers) {
  if (typeof mothers == "string") {
    return hiMom(mothers);
  } else if (mothers instanceof Array) {
    return mothers.map((mother) => hiMom(mother?.name, mother?.lang)).join(" ");
  }

  throw new Error("Mother type not supported, but hi mom anyway!");
}

exports.hiMom = hiMom;
exports.hiMoms = hiMoms;
